### DISCUSSION TITLE
How to merge two graphs

### DISCUSSION THREAD

**Discussion by gcorso** [pyg-team/pytorch_geometric#6311] [Q&A] [answered]
I have two heterogeneous graphs (HeteroData) with disjoint node and edge classes that I need to merge into a single graph (to then create edges between the different type of nodes).

Right now I have tried the following code:
```
complex_graph = copy.deepcopy(receptor_graph)     # contains "receptor" node type
complex_graph['ligand'] = copy.deepcopy(ligand_graph['ligand'])
complex_graph['ligand', 'lig_bond', 'ligand'] = copy.deepcopy(ligand_graph['ligand', 'lig_bond', 'ligand'])
```

However this does not work because `complex_graph['ligand']` results being of class `dict` instead of class `torch_geometric.data.storage.NodeStorage`  as `complex_graph['receptor']`. I have also tried using `get_node_store` without success (same issue).

How can I properly merge the two graphs? 
Thank you very much for the help!

**Comment by rusty1s** [comment, marked as answer]
We currently don't provide any dedicated function for this, so this gets a bit cumbersome. How about
```python
new_graph = copy.copy(receptor_graph)
for type in ligrand_graph.node_types + ligrand_graph.edge_types:
    for key, value in ligrand_graph[type].items():
        new_graph[type][key] = value

**Comment by gcorso** [comment]
This worked!
Thank you very much for the answer and, as always, for the terrific work Matthias!

**Reply by rusty1s** [reply]
:) Also added this functionality into PyG directly, see https://github.com/pyg-team/pytorch_geometric/pull/6313. You should now be able to do:
```python
receptor_graph.update(ligrand_grad)
```
